src/reports/library/report.hpp

```diff
@@ -52,6 +34,19 @@ class Report {
     virtual void end_iteration(double timestep);
     virtual void flush(double time);
     void refresh_pointers(refresh_function_t refresh_function);
-
     void set_max_buffer_size(size_t buffer_size);
+
+  protected:
+    using nodes_t = std::map<uint64_t, Node>;
```

Comment:
If the iterator ordering doesn't matter, consider using `std::unordered_map` instead.

Reply:
I changed the report map and the communicator map to unordered.
However  in the case of the node map, although is not strictly necessary, I prefer to keep it like this for now until we release the library, because for testing/debugging it is easier to read the reports if the gids are ordered.